src/mpid/ch4/src/ch4r_win.h

```diff
@@ -1269,6 +1269,15 @@ static inline int MPIDI_CH4R_mpi_win_shared_query(MPIR_Win * win,
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_CH4R_MPI_WIN_SHARED_QUERY);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_CH4R_MPI_WIN_SHARED_QUERY);
 
+    /* When only single process exists on the node, should only query
+     * MPI_PROC_NULL or local process. Thus, return local window's info. */
+    if (win->comm_ptr->node_comm == NULL) {
+        *size = win->size;
+        *disp_unit = win->disp_unit;
+        *((void **) baseptr) = win->base;
+        goto fn_exit;
+    }
+
```

Comment:
This patch (`ch4/rma: fix win_shared_query when no shm env`) should be squashed with the appropriate patch before merging.

Reply:
Make sense. I will squash it with patch `ch4/rma: define win alloc common routine using shm symheap`, where `win_allocate` starts creating shared memory.